include/aws/common/json.h

```diff
@@ -0,0 +1,325 @@
+ * @param key The key that the aws_json_value is at. Is not case sensitive.
+ * @return The aws_json_value at the given key, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_object_get_insensitive(const struct aws_json_value *object,
+                                                       const struct aws_byte_cursor *cursor,
+                                                       const struct aws_allocator *allocator);
+
+/**
+ * Checks if there is a aws_json_value at the given key.
+ * @param object The object aws_json_value you want to check a key in.
+ * @param key The key that you want to check. Is case sensitive.
+ * @return AWS_OP_SUCCESS if a aws_json_value is found, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_has(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                        const struct aws_allocator *allocator);
+
+/**
+ * Removes the aws_json_value at the given key.
+ * @param object The object aws_json_value you want to remove a aws_json_value in.
+ * @param key The key that the aws_json_value is at. Is case sensitive.
+ * @return AWS_OP_SUCCESS if the aws_json_value was removed, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_remove(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                           const struct aws_allocator *allocator);
+// ====================
+
+// ====================
+// Array API
+
+/**
+ * Adds a aws_json_value to the given array aws_json_value.
+ * @param array The array aws_json_value you want to add an aws_json_value to.
+ * @param node The aws_json_value you want to add.
+ * @return AWS_OP_SUCCESS if adding the aws_json_value was successful, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_array_add(const struct aws_json_value *array, const struct aws_json_value *node);
+
+/**
+ * Returns the aws_json_value at the given index in the array aws_json_value.
+ * @param array The array aws_json_value.
+ * @param index The index of the aws_json_value you want to access.
+ * @return A pointer to the aws_json_value at the given index in the array, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_array_get(const struct aws_json_value *array, const size_t index);
+
+/**
+ * Returns the number of items in the array aws_json_value.
+ * @param array The array aws_json_value.
+ * @return The number of items in the array_json_node.
+ */
+AWS_COMMON_API
+int aws_json_array_get_count(const struct aws_json_value *array);
+
+/**
+ * Removes the aws_json_value at the given index in the array aws_json_value.
+ * @param array The array aws_json_value.
+ * @param index The index containing the aws_json_value you want to remove.
+ * @return AWS_OP_SUCCESS if the aws_json_value at the index was removed, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_array_remove(const struct aws_json_value *array, const size_t index);
```

Comment:
I'm not sure. The case I can see for having it is for data manipulation, like if you need to update a value in the array. Outside of that though, I do not think it would be needed.